test/com/google/enterprise/adaptor/documentum/DocumentumAdaptorTest.java

```diff
@@ -5245,34 +5245,99 @@ public void testModifiedFolderUpdatedDocumentsQuery() throws Exception {
         makeExpectedDocIds(folder, folder, "bbb"),
         new Checkpoint(MAR_1970, DOCUMENT.pad("bbb")));
 
-    StringBuilder query = new StringBuilder();
-    query.append("SELECT i_chronicle_id, r_object_id, object_name, ")
-    .append("DATETOSTRING_LOCAL(r_modify_date, ''yyyy-mm-dd hh:mi:ss'') ")
-    .append("AS r_modify_date_str ")
-    .append("FROM dm_sysobject ")
-    .append("WHERE FOLDER(''/FFF1'',descend) ")
-    .append("AND ((r_modify_date = DATE(''{0}'',''yyyy-mm-dd hh:mi:ss'') ")
-    .append("AND r_object_id > ''{1}'') ")
-    .append("OR (r_modify_date > DATE(''{0}'',''yyyy-mm-dd hh:mi:ss''))) ")
-    .append("UNION ")
-    .append("SELECT i_chronicle_id, r_object_id, object_name, ")
-    .append("DATETOSTRING_LOCAL(r_modify_date, ''yyyy-mm-dd hh:mi:ss'') ")
-    .append("AS r_modify_date_str ")
-    .append("FROM dm_sysobject ")
-    .append("WHERE r_object_id = ''")
-    .append(DOCUMENT.pad("aaa"))
-    .append("'' ")
-    .append("ORDER BY 3 ASC,2 ASC ENABLE(RETURN_TOP 500)");
-
     checkModifiedDocIdsPushed(
-        ImmutableMap.of("documentum.updatedDocumentsQuery", query),
+        ImmutableMap.of("documentum.modifiedDocumentsQuery", query),
         startPaths(folder),
         new Checkpoint(FEB_1970, DOCUMENT.pad("aaa")),
-        makeExpectedDocIds(folder, folder, "aaa", "bbb"),
+        makeExpectedDocIds(folder, expected),
         new Checkpoint(MAR_1970, DOCUMENT.pad("bbb")));
   }
 
   @Test
+  public void testModifiedDocumentsQuery() throws Exception {
+    String folder = "/FFF1/FFF2";
+    String query = "SELECT i_chronicle_id, r_object_id, object_name, "
+        +"r_modify_date, "
+        + "DATETOSTRING_LOCAL(r_modify_date, ''yyyy-mm-dd hh:mi:ss'') "
+        +"AS r_modify_date_str "+"FROM dm_sysobject "
+        +"WHERE FOLDER(''/FFF1'',descend) "
+        +"AND ((r_modify_date = DATE(''{0}'',''yyyy-mm-dd hh:mi:ss'') "
+        +"AND r_object_id > ''{1}'') "
+        +"OR (r_modify_date > DATE(''{0}'',''yyyy-mm-dd hh:mi:ss''))) "
+        +"UNION "
+        +"SELECT i_chronicle_id, r_object_id, object_name, r_modify_date, "
+        +"DATETOSTRING_LOCAL(r_modify_date, ''yyyy-mm-dd hh:mi:ss'') "
+        +"AS r_modify_date_str "
+        +"FROM dm_sysobject "
+        +"WHERE r_object_id = ''" + DOCUMENT.pad("aaa") + "'' "
+        +"ORDER BY 4 ASC,2 ASC ENABLE(RETURN_TOP 500)";
+
+    testModifiedFolderModifiedDocumentsQuery(query, "aaa", folder, "bbb");
```

Comment:
I don't like the duplication of the test data here, such as "folder" and "aaa", or the use of varargs. Those dislikes are at odds with each other. The former is harder and the latter is stronger, so I would prefer to say makeExpectedDocIds(folder, folder, "aaa", "bbb") here.